crates/sui-config/src/node.rs

```diff
@@ -455,6 +463,17 @@ impl ConsensusConfig {
     pub fn narwhal_config(&self) -> &NarwhalParameters {
         &self.narwhal_config
     }
+
+    pub fn db_retention_epochs(&self) -> u64 {
+        self.db_retention_epochs.unwrap_or(1)
+    }
+
+    pub fn db_pruner_period(&self) -> Duration {
```

Comment:
Can we check on startup and every epoch instead of hourly?

Reply:
Basically I've used the hourly approach mostly as a generic retry mechanism - instead of trying to build a more exclusive retry logic in the code. The boundaries are calculated on every epoch change anyways.